client/gutenberg/editor/components/post-publish-button-or-toggle/index.jsx

```diff
@@ -0,0 +1,223 @@
+		 * 	 we decided to take into account the viewport in that case.
+		 *  	 See: https://github.com/WordPress/gutenberg/issues/10475
+		 *
+		 * 2) Then, in small viewports, we'll show a TOGGLE.
+		 *
+		 * 3) Finally, we'll use the publish sidebar status to decide:
+		 *
+		 * - if it is enabled, we show a TOGGLE
+		 * - if it is disabled, we show a BUTTON
+		 */
+		if (
+			isPublished ||
+			( isScheduled && isBeingScheduled ) ||
+			( isPending && ! hasPublishAction && ! isLessThanMediumViewport )
+		) {
+			component = IS_BUTTON;
+		} else if ( isLessThanMediumViewport ) {
+			component = IS_TOGGLE;
+		} else if ( isPublishSidebarEnabled ) {
+			component = IS_TOGGLE;
+		} else {
+			component = IS_BUTTON;
+		}
+
+		return (
+			<Fragment>
+				<PostPublishButton
+					forceIsDirty={ forceIsDirty }
+					forceIsSaving={ forceIsSaving }
+					isOpen={ isPublishSidebarOpened }
+					isToggle={ component === IS_TOGGLE }
+					onToggle={ togglePublishSidebar }
+				/>
+				{ showEmailVerificationNotice && (
+					<VerifyEmailDialog onClose={ this.closeVerifyEmailDialog } />
+				) }
+			</Fragment>
+		);
+	}
+}
+
+const applyWithSelect = withSelect( select => {
+	const {
+		getCurrentPost,
+		isEditedPostBeingScheduled,
+		isCurrentPostPending,
+		isCurrentPostPublished,
+		isPublishSidebarEnabled,
+		isCurrentPostScheduled,
+		isPostSavingLocked,
+	} = select( 'core/editor' );
+
+	const { isPublishSidebarOpened } = select( 'core/edit-post' );
+
+	const isPublished = isCurrentPostPublished();
+	const isBeingScheduled = isEditedPostBeingScheduled();
+
+	let verificationNoticeLabel;
+	if ( isPublished ) {
+		verificationNoticeLabel = translate( 'To update, check your email and confirm your address.' );
+	} else if ( isBeingScheduled ) {
+		verificationNoticeLabel = translate(
+			'To schedule, check your email and confirm your address.'
+		);
+	} else {
+		verificationNoticeLabel = translate( 'To publish, check your email and confirm your address.' );
+	}
+
+	return {
+		hasPublishAction: get( getCurrentPost(), [ '_links', 'wp:action-publish' ], false ),
+		isBeingScheduled,
+		isPending: isCurrentPostPending(),
+		isPublished,
+		isPublishSidebarEnabled: isPublishSidebarEnabled(),
+		isPublishSidebarOpened: isPublishSidebarOpened(),
+		isScheduled: isCurrentPostScheduled(),
+		isPostSavingLocked: isPostSavingLocked(),
+		verificationNoticeLabel,
+	};
+} );
+
+const applyWithDispatch = withDispatch( dispatch => {
+	const { togglePublishSidebar } = dispatch( 'core/edit-post' );
+	const { createWarningNotice } = dispatch( 'core/notices' );
+	const { lockPostSaving, unlockPostSaving } = dispatch( 'core/editor' );
+	return {
+		togglePublishSidebar,
+		createWarningNotice,
+		lockPostSaving,
+		unlockPostSaving,
+	};
+} );
+
+export default compose(
+	applyWithSelect,
+	applyWithDispatch,
+	withViewportMatch( { isLessThanMediumViewport: '< medium' } ),
+	withState( {
+		showEmailVerificationNotice: false,
```

Comment:
What would be the advantage of using `withState` instead of `this.state`?

Reply:
Actually, none on this particular component. 

I think `withState` is great for functional components, because it allows to have state on them, without needing to transform them to classes. That's why I like to use `withState`.

But this component is already a class component so `withState` is not adding any specific benefit. 

Do you think I should use `this.state`?

Reply:
That's interesting, I didn't really know of `withState`.

To be honest, though, in Calypso we are perfectly fine in using the classical component state, and `withState` feels a bit out of place to me.

Also, as I said [elsewhere](https://github.com/Automattic/wp-calypso/pull/29594/files#r242898999), please change this variable name because it's rather confusing. 🙂

```diff
- showEmailVerificationNotice
+ showEmailVerificationDialog
```